Even with 4b146a79fe78dce6f08a90a2be2becd263e84a2b for me the following test fails:


     def test_issue_15082():
         exprs = [1]
         cse_eq = cse(exprs)[1][0]
         assert cse_eq == 1
         exprs = 1
         cse_eq = cse(exprs)[1][0]
         assert cse_eq == 1

fails with:

> python3 bin/test sympy/simplify/tests/test_cse.py 
> =============================================================================================== test process starts ===============================================================================================
> executable:         /usr/bin/python3  (3.6.5-final-0) [CPython]
> architecture:       64-bit
> cache:              yes
> ground types:       gmpy 2.0.8
> numpy:              1.13.3
> random seed:        41199369
> hash randomization: on (PYTHONHASHSEED=2345804990)
> 
> sympy/simplify/tests/test_cse.py[43] .........f.fff....................f.......E                                                                                                                             [FAIL]
> 
> ___________________________________________________________________________________________________________________________________________________________________________________________________________________
> ________________________________________________________________________________ sympy/simplify/tests/test_cse.py:test_issue_15082 ________________________________________________________________________________
> Traceback (most recent call last):
>   File "/home/moggi/devel/sympy/sympy/simplify/tests/test_cse.py", line 545, in test_issue_15082
>     cse_eq = cse(exprs)[1][0]
>   File "/home/moggi/devel/sympy/sympy/simplify/cse_main.py", line 697, in cse
>     for e in exprs:
> TypeError: 'int' object is not iterable

I don't see how this could actually ever work as the only line that would make the exprs input an iterable object is line 692 in sympy/simplify/cse_main.py

Maybe I'm missing something.

--- a/sympy/simplify/cse_main.py
+++ b/sympy/simplify/cse_main.py
@@ -688,6 +688,9 @@ def cse(exprs, symbols=None, optimizations=None, postprocess=None,
     from sympy.matrices import (MatrixBase, Matrix, ImmutableMatrix,
                                 SparseMatrix, ImmutableSparseMatrix)
 
+    if isinstance(exprs, (int, float)):
+        exprs = sympify(exprs)
+
     # Handle the case if just one expression was passed.
     if isinstance(exprs, (Basic, MatrixBase)):
         exprs = [exprs]

--- a/sympy/simplify/tests/test_cse.py
+++ b/sympy/simplify/tests/test_cse.py
@@ -74,6 +74,9 @@ def test_cse_single():
 
     subst42, (red42,) = cse([42])  # issue_15082
     assert len(subst42) == 0 and red42 == 42
+    subst_half, (red_half,) = cse([0.5])
+    assert len(subst_half) == 0 and red_half == 0.5
+
 
 def test_cse_single2():
     # Simple substitution, test for being able to pass the expression directly
@@ -86,6 +89,8 @@ def test_cse_single2():
 
     subst42, (red42,) = cse(42)  # issue 15082
     assert len(subst42) == 0 and red42 == 42
+    subst_half, (red_half,) = cse(0.5)  # issue 15082
+    assert len(subst_half) == 0 and red_half == 0.5
 
 def test_cse_not_possible():
     # No substitution possible.

